### PR TITLE
fix: Intel x86_64 macOS issue with wrapt 1.17

### LINKED ISSUE #5655: [BUG] Unable to run Phoenix-Arize on Mac pro 2019 intel-i9
### Issue: Metaclass conflict when using the phoenix library

I'm encountering an issue when trying to use the phoenix library. Upon importing the library, I get the following error:

```
Traceback (most recent call last):
  File "/Users/<user>/<folder>/<python.py>", line 1, in <module>
    import phoenix
  File "/Users/<user>/<folder>/.venv/lib/python3.12/site-packages/phoenix/__init__.py", line 12, in <module>
    from .session.session import (
  File "/Users/<user>/<folder>/.venv/lib/python3.12/site-packages/phoenix/session/session.py", line 33, in <module>
    from phoenix.core.model_schema_adapter import create_model_from_inferences
  File "/Users/<user>/<folder>/.venv/lib/python3.12/site-packages/phoenix/core/model_schema_adapter.py", line 11, in <module>
    from phoenix.core.model_schema import Embedding, Model, RetrievalEmbedding, Schema
  File "/Users/<user>/<folder>/.venv/lib/python3.12/site-packages/phoenix/core/model_schema.py", line 545, in <module>
    class ModelData(ObjectProxy, ABC):  # type: ignore
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases

```
### What I’ve Tried
- Tried installing different versions of arize-phoenix.
- Reinstalled pipenv, pyenv, and Python.
- Created a fresh virtual environment and only installed arize-phoenix.

### Environment Details
- MacOS: MacBook Pro 2019 (Intel i9)
- Python: 3.12.6
- IDE: VS Code
- Working Setup: The issue does not occur on a MacBook M1 with the same library version and environment setup.

### Code Causing the Issue
```
class ModelData(ObjectProxy, ABC):  # type: ignore
    """pd.DataFrame or pd.Series wrapped with extra functions and metadata."""

    def __init__(
        self,
        data: DataFrameOrSeries,
        /,
        _model: Optional["ProxyType[Model]"] = None,
    ) -> None:
        super().__init__(data)
        self._self_model = _model

    @property
    @abstractmethod
    def null_value(self) -> Any: ...

    def __getitem__(self, key: Any) -> Any:
        if _is_column_key(key):
            if isinstance(key, Column):
                return key(self)
            if self._self_model is None or self.empty:
                return self.null_value
            model = cast(Model, self._self_model)
            return model[key](self)
        return super().__getitem__(key)

```

### Additional Context
Previously I faced the same issue but downgrading to 5.6.0 everything worked fine. Now after the 6.1.0 update I can not run it on my environment.

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #5655 

https://github.com/GrahamDumpleton/wrapt/issues/276

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies = [
   "pyarrow",
   "typing-extensions>=4.6",
   "scipy",
-  "wrapt",
+  "wrapt<1.17",
```
Comment (axiomofjoy) [inline pyproject.toml:35]: This seems to be related to an issue reported [here](https://github.com/Arize-ai/phoenix/discussions/5532), but I haven't been able to reproduce locally.

Comment (axiomofjoy) [reply pyproject.toml:35]: Is it just an issue on Intel?

Comment (RogerHYang) [reply pyproject.toml:35]: yea, it's just intel mac. so weird

Comment (RogerHYang) [reply pyproject.toml]: probably related to [this](https://github.com/GrahamDumpleton/wrapt/issues/276)

**Pull request merged**